internal/cli/atlas/privateendpoints/gcp/interfaces/delete.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interfaces
+
+import (
+	"context"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.GlobalOpts
+	*cli.DeleteOpts
+	privateEndpointServiceID string
+	privateEndpointGroupID   string
+	store                    store.InterfaceEndpointDeleter
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *DeleteOpts) Run() error {
+	return opts.Delete(opts.store.DeleteInterfaceEndpoint, opts.ConfigProjectID(), provider, opts.privateEndpointServiceID, opts.privateEndpointGroupID)
+}
+
+// mongocli atlas privateEndpoint(s) gcp interface(s) delete <endpointGroupId> --endpointServiceId endpointServiceId [--projectId projectId].
+func DeleteBuilder() *cobra.Command {
+	opts := &DeleteOpts{
+		DeleteOpts: cli.NewDeleteOpts("Interface endpoint '%s' deleted\n", "Interface endpoint not deleted"),
+	}
+	cmd := &cobra.Command{
+		Use:     "delete <endpointGroupId>",
```

Comment:
[q] I see later the opt is `privateEndpointServiceID` - should we use that name everywhere?

Reply:
There are two similar but distinct concepts: that of the endpoint service and that of the endpoint itself.

At the API level the DELETE method, across all cloud providers, uses path:

  /groups/{GROUP-ID}/privateEndpoint/{CLOUD-PROVIDER}/endpointService/{ENDPOINT-SERVICE-ID}/endpoint/{ENDPOINT-ID}

But in mongocli the terminology is different across cloud providers, and from my understanding of the spec doc for GCP endpoint interfaces:

  The endpointGroupId args maps to ENDPOINT-ID (the last component of the path)
  The --endpointServiceId flag maps to ENDPOINT-SERVICE-ID

So, privateEndpointServiceID in opt is not the same as endpointGroupId.

However, I notice now that privateEndpointGroupID in opt does not belong there (ie. it's redundant) so I will remove that.